php-checks/src/main/java/org/sonar/php/checks/HardCodedCredentialsInVariablesAndUrisCheck.java

```diff
@@ -106,7 +106,7 @@ private void checkForCredentialUri(LiteralTree literal) {
       String userInfo = uri.getUserInfo();
 
       String[] splitUserInfo = userInfo.split(":");
-      if (splitUserInfo.length == 1 || splitUserInfo[0].equals(splitUserInfo[1]) || isCommonTestCredential(userInfo)) {
+      if (splitUserInfo.length < 2 || splitUserInfo[0].equals(splitUserInfo[1]) || isCommonTestCredential(userInfo)) {
```

Comment:
I would also add a test case which work with this new code but fail with the previous version.